### PR TITLE
allow to print DecisionTreeClassifier as text or svg

### PR DESCRIPTION
**Pull request by kmike**
Currently output looks like this:

```
petal width <= 2.450  (33.3%)  ---> [1.000, 0.000, 0.000]
petal width > 2.450  (66.7%)
    petal length <= 1.750  (36.0%)
        petal width <= 4.950  (32.0%)  ---> [0.000, 0.979, 0.021]
        petal width > 4.950  (4.0%)  ---> [0.000, 0.333, 0.667]
    petal length > 1.750  (30.7%)
        petal width <= 4.850  (2.0%)  ---> [0.000, 0.333, 0.667]
        petal width > 4.850  (28.7%)  ---> [0.000, 0.000, 1.000]
```

### CODE REVIEW

File: eli5/sklearn/explain_weights.py [hunk 1]
```diff
@@ -28,11 +28,13 @@
     RandomForestClassifier,
     ExtraTreesClassifier
 )
-from sklearn.tree import DecisionTreeClassifier
+from sklearn.tree import DecisionTreeClassifier, DecisionTreeRegressor
+from sklearn.tree.export import export_graphviz
```
Comment (lopuhin) [inline eli5/sklearn/explain_weights.py]: This is not used? Didn't know about that function!

File: eli5/sklearn/treeinspect.py [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+        if not node['is_leaf']:
+            feat_id = node['feature_id']
+            if feature_names is None:
+                node['feature_name'] = "x%s" % feat_id
+            else:
+                node['feature_name'] = feature_names[feat_id]
+
+
+def _node2dict(tree, node_id):
+    is_leaf = tree.children_left[node_id] == _tree.TREE_LEAF
```
Comment (lopuhin) [inline eli5/sklearn/treeinspect.py]: can a left child be a leaf while right non-leaf?

**Comment by lopuhin** [comment]
Cool, that's much more informative than just feature weights!

File: eli5/sklearn/explain_weights.py (hunk 1 above)
Comment (kmike) [inline eli5/sklearn/explain_weights.py]: yeah, it is unused; I want to use it for HTML export

File: eli5/sklearn/treeinspect.py (hunk 2 above)
Comment (kmike) [inline eli5/sklearn/treeinspect.py]: the code above is a way to check if a current node is a leaf, not if its left child is a leaf; I copy-pasted it from export_graphviz

Comment (lopuhin) [inline eli5/sklearn/treeinspect.py]: Right, now I understand, thanks!
